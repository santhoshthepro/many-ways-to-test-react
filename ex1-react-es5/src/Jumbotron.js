var React = require('react');

var Jumbotron = React.createClass({
  render: function() {
    return (
      <div className="jumbotron">
        <div className="ghosted">
          <h1>Dogs: The Musical</h1>
          <p>“It’ll have you barking in your seat!”</p>
          <p><a href="#" className="btn btn-primary">Buy Tickets</a></p>
        </div>
      </div>
    );
  }
});

module.exports = Jumbotron;
