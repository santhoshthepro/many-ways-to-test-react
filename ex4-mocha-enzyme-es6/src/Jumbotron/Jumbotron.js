import React from 'react';

const Jumbotron = (() => {
  return (
    <div className="jumbotron">
      <div className="ghosted">
        <h1>Dogs: The Musical</h1>
        <p>“It’ll have you barking in your seat!”</p>
        <p><a href="#" className="btn btn-primary">Buy Tickets</a></p>
      </div>
    </div>
  );
});

export default Jumbotron;
