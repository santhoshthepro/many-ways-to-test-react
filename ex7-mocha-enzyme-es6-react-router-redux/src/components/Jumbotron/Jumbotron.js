import React from 'react';
import { Link } from 'react-router';

const Jumbotron = (() => {
  return (
    <div className="jumbotron">
      <div className="ghosted">
        <h1>Dogs: The Musical</h1>
        <p>“It’ll have you barking in your seat!”</p>
        <p><Link to="/tickets" className="btn btn-primary btn-lg">Buy Tickets</Link></p>
      </div>
    </div>
  );
});

export default Jumbotron;
