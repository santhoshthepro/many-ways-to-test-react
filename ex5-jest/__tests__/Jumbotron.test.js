import React from 'react';
import renderer from 'react-test-renderer';
import Jumbotron from '../src/Jumbotron/Jumbotron';

test('Jumbotron rendered static HTML', () => {
  const component = renderer.create(
    <Jumbotron />
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
