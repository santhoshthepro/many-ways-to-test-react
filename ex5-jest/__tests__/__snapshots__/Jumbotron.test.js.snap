exports[`test Jumbotron rendered static HTML 1`] = `
<div
  className="jumbotron">
  <div
    className="ghosted">
    <h1>
      Dogs: The Musical
    </h1>
    <p>
      “It’ll have you barking in your seat!”
    </p>
    <p>
      <a
        className="btn btn-primary"
        href="#">
        Buy Tickets
      </a>
    </p>
  </div>
</div>
`;
